File: examples/weather-mcp/src/main/java/com/example/weather/model/GetWeatherResponse.java
/*
 * Copyright (c) 2026 Konstantin Pavlov/IT Staff and contributors.
 */

package com.example.weather.model;

import com.fasterxml.jackson.annotation.JsonClassDescription;
import com.fasterxml.jackson.annotation.JsonPropertyDescription;
import com.fasterxml.jackson.annotation.JsonTypeName;

@JsonClassDescription("Current weather observation for a city.")
@JsonTypeName("GetWeatherResponse")
public record GetWeatherResponse(
        @JsonPropertyDescription("City name") String city,
        @JsonPropertyDescription("Weather condition") String condition,
        @JsonPropertyDescription("Temperature in the response unit") double temperature,
        @JsonPropertyDescription("Temperature unit") TemperatureUnit unit,
        @JsonPropertyDescription("Relative humidity percentage") int humidity,
        @JsonPropertyDescription("Wind speed in km/h") double windSpeed) {
}

File: examples/weather-mcp-kotlin/src/main/kotlin/com/example/weather/model/GetWeatherResponse.kt
package com.example.weather.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import me.kpavlov.kt.schema.Description

@Serializable
@SerialName("GetWeatherResponse")
data class GetWeatherResponse(
    @Description("City name")
    val city: String,
    @Description("Weather condition")
    val condition: String,
    @Description("Temperature in the response unit")
    val temperature: Double,
    @Description("Temperature unit")
    val temperatureUnit: TemperatureUnit,
    @Description("Relative humidity percentage")
    val humidity: Int,
    @Description("Wind speed in km/h")
    val windSpeed: Double,
)
